Add molecule testing to template

DrewHarvard opened the pull request:
Addition of molecule structure we have adapted for testing ansible roles.
This will act as a basic template for the user to add molecule tests. Depending on the complexity of the role and what you want to test there may be more thats required.

Resolves: DVOP-1417

DrewHarvard committed:
Add molecule testing to template

Addition of molecule structure we have adapted for testing ansible roles.
This will act as a basic template for the user to add molecule tests. Depending on the complexity of the role and what you want to test there may be more thats required.

Resolves: DVOP-1417

DrewHarvard committed:
Rename roles and tasks to use hyphened approach for naming

marcransome commented on role-name/molecule/roles/converge.yml:
```diff
@@ -0,0 +1,7 @@
+- name: Converge
+  hosts: all
+  tasks:
+    - name: "Include my-new-role"
```
It's worth considering whether we standardise on placeholders throughout this repository to ensure consistency and make it easier to highlight to the reader what needs to be specified in multiple places, so `Include my-new-role` here could be replaced with `<role-name>` as is used in other files.

marcransome commented on role-name/molecule/roles/converge.yml:
```diff
@@ -0,0 +1,7 @@
+  tasks:
+    - name: "Include my-new-role"
+      include_role:
+        name: "my-new-role"
```
As above, we could make this `<role-name>` for consistency.

marcransome commented on role-name/molecule/roles/molecule.yml:
```diff
@@ -0,0 +1,30 @@
+    image: centos:8
+    pre_build_image: true
+lint:
+  'set -e ansible-lint'
```
The `set -e` command isn't strictly required here as the exit code of the last command will be used and given that we're only executing the single command `ansible-lint`. Having said that, if roles created from this template need to extend the linting stage by introducing additional commands we would want the first command that produces a non-zero exit code in the context of a list of commands to be used. I'd suggest we keep `set -e`, but use the pipe character to indicate a block-style literal:

```yaml
lint: |
  set -e
  ansible-lint
```

This makes it clearer that `set -e` is a distinct command by itself, and also indicates how we write multiple commands on separate lines which may be useful to anyone using this template repository, if additional commands need to be added.

marcransome commented on role-name/molecule/roles/verify.yml:
```diff
@@ -0,0 +1,9 @@
+  tasks:
+    - name: Test Description
+      include_role:
+        name: test-role-name
```
As above, we could add a placeholder here such as `test-<role-name>` to ensure consistency.

marcransome commented on role-name/tests/test.yml:
```diff
@@ -2,4 +2,4 @@
   roles:
-    - role_name
\ No newline at end of file
+    - role-name
```
As above, this would be better written as a placeholder: `<role-name>`.

DrewHarvard committed:
Address suggested changes

Follow a suitable pattern for declaring the role-name to make it easier for the user to understand and keep consistent throughout.
